### ISSUE TITLE
make install DESTDIR=$PKG

### ISSUE THREAD

**Issue by 0pLuS0** [jurplel/qView#352]
**Environment:**
- OS: OS: Slackware 14.2 x86_64
- qView version: 3.0
qt5-5.9.8_nowebengine

**Information:**
The source code doesn't seem to respect; ```make install DESTDIR=$PKG```

This is how I'm compiling it from a Slackware build script;

cd $TMP/$PRGNAM-$VERSION
qmake-qt5
make
make install DESTDIR=$PKG

Arch Linux I noticed uses this;

	cd qView
	qmake
	make

	cd qView
	make INSTALL_ROOT="$pkgdir/" install

So then I changed my build script to now be like this;

cd $TMP/$PRGNAM-$VERSION
qmake-qt5
make
make INSTALL_ROOT="$PKG" install

This is the contents of the slackware package, can you please tell me if this looks complete/correct?

./
install/
install/slack-desc
usr/
usr/bin/
usr/bin/qview
usr/share/
usr/share/applications/
usr/share/applications/qView.desktop
usr/share/licenses/
usr/share/licenses/qview/
usr/share/licenses/qview/LICENSE
usr/share/icons/
usr/share/icons/hicolor/
usr/share/icons/hicolor/16x16/
usr/share/icons/hicolor/16x16/apps/
usr/share/icons/hicolor/16x16/apps/qview.png
usr/share/icons/hicolor/32x32/
usr/share/icons/hicolor/32x32/apps/
usr/share/icons/hicolor/32x32/apps/qview.png
usr/share/icons/hicolor/64x64/
usr/share/icons/hicolor/64x64/apps/
usr/share/icons/hicolor/64x64/apps/qview.png
usr/share/icons/hicolor/scalable/
usr/share/icons/hicolor/scalable/apps/
usr/share/icons/hicolor/scalable/apps/qview.svg
usr/share/icons/hicolor/128x128/
usr/share/icons/hicolor/128x128/apps/
usr/share/icons/hicolor/128x128/apps/qview.png
usr/share/icons/hicolor/256x256/
usr/share/icons/hicolor/256x256/apps/
usr/share/icons/hicolor/256x256/apps/qview.png

```THANKS```

**Comment by jurplel** [comment]
I'm not familiar with slackware but it looks like all of the important files are there.

**Comment by 0pLuS0** [comment]
Ok...

By the way, I noticed when I bring up the About menu or the 'Show file info', there is no way to close these windows, I assumed when I click back in the main UI of qView they would disappear, but they don't. 

So I have a keyboard binding I use to close apps instead of closing them through the UI, to make this process faster and this is what I have to do in order to get these windows closed...

Hmm

Is this an issue you are aware of, would you like to make a Issue for this?

THANKS

**Comment by jurplel** [comment]
What desktop environment are you using?

**Comment by 0pLuS0** [comment]
Openbox 3.6.1

**Comment by jurplel** [comment]
Can you try it on the latest master and see if it's still an issue?

**Comment by 0pLuS0** [comment]
https://github.com/jurplel/qView.git ?

**Comment by jurplel** [comment]
Yes

**Comment by 0pLuS0** [comment]
It failed to compile;

qView-3.0/.git/branches/
Info: creating stash file /tmp/sar/qView-3.0/.qmake.stash
/usr/lib64/qt5/bin/uic src/mainwindow.ui -o intermediate/ui_mainwindow.h
/usr/lib64/qt5/bin/uic src/qvoptionsdialog.ui -o intermediate/ui_qvoptionsdialog.h
/usr/lib64/qt5/bin/uic src/qvaboutdialog.ui -o intermediate/ui_qvaboutdialog.h
/usr/lib64/qt5/bin/uic src/qvwelcomedialog.ui -o intermediate/ui_qvwelcomedialog.h
/usr/lib64/qt5/bin/uic src/qvinfodialog.ui -o intermediate/ui_qvinfodialog.h
/usr/lib64/qt5/bin/uic src/qvshortcutdialog.ui -o intermediate/ui_qvshortcutdialog.h
g++ -c -pipe -O2 -fPIC -std=gnu++11 -Wall -W -D_REENTRANT -fPIC -DVERSION=3.0 -DQT_DEPRECATED_WARNINGS -DQT_NO_FOREACH -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I. -isystem /usr/include/qt5 -isystem /usr/include/qt5/QtWidgets -isystem /usr/include/qt5/QtGui -isystem /usr/include/qt5/QtNetwork -isystem /usr/include/qt5/QtCore -Iintermediate -isystem /usr/include/libdrm -Iintermediate -I/usr/lib64/qt5/mkspecs/linux-g++ -o intermediate/main.o src/main.cpp
g++ -c -pipe -O2 -fPIC -std=gnu++11 -Wall -W -D_REENTRANT -fPIC -DVERSION=3.0 -DQT_DEPRECATED_WARNINGS -DQT_NO_FOREACH -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I. -isystem /usr/include/qt5 -isystem /usr/include/qt5/QtWidgets -isystem /usr/include/qt5/QtGui -isystem /usr/include/qt5/QtNetwork -isystem /usr/include/qt5/QtCore -Iintermediate -isystem /usr/include/libdrm -Iintermediate -I/usr/lib64/qt5/mkspecs/linux-g++ -o intermediate/mainwindow.o src/mainwindow.cpp
g++ -c -pipe -O2 -fPIC -std=gnu++11 -Wall -W -D_REENTRANT -fPIC -DVERSION=3.0 -DQT_DEPRECATED_WARNINGS -DQT_NO_FOREACH -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I. -isystem /usr/include/qt5 -isystem /usr/include/qt5/QtWidgets -isystem /usr/include/qt5/QtGui -isystem /usr/include/qt5/QtNetwork -isystem /usr/include/qt5/QtCore -Iintermediate -isystem /usr/include/libdrm -Iintermediate -I/usr/lib64/qt5/mkspecs/linux-g++ -o intermediate/qvgraphicsview.o src/qvgraphicsview.cpp
g++ -c -pipe -O2 -fPIC -std=gnu++11 -Wall -W -D_REENTRANT -fPIC -DVERSION=3.0 -DQT_DEPRECATED_WARNINGS -DQT_NO_FOREACH -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I. -isystem /usr/include/qt5 -isystem /usr/include/qt5/QtWidgets -isystem /usr/include/qt5/QtGui -isystem /usr/include/qt5/QtNetwork -isystem /usr/include/qt5/QtCore -Iintermediate -isystem /usr/include/libdrm -Iintermediate -I/usr/lib64/qt5/mkspecs/linux-g++ -o intermediate/qvoptionsdialog.o src/qvoptionsdialog.cpp
g++ -c -pipe -O2 -fPIC -std=gnu++11 -Wall -W -D_REENTRANT -fPIC -DVERSION=3.0 -DQT_DEPRECATED_WARNINGS -DQT_NO_FOREACH -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I. -isystem /usr/include/qt5 -isystem /usr/include/qt5/QtWidgets -isystem /usr/include/qt5/QtGui -isystem /usr/include/qt5/QtNetwork -isystem /usr/include/qt5/QtCore -Iintermediate -isystem /usr/include/libdrm -Iintermediate -I/usr/lib64/qt5/mkspecs/linux-g++ -o intermediate/qvapplication.o src/qvapplication.cpp
g++ -c -pipe -O2 -fPIC -std=gnu++11 -Wall -W -D_REENTRANT -fPIC -DVERSION=3.0 -DQT_DEPRECATED_WARNINGS -DQT_NO_FOREACH -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I. -isystem /usr/include/qt5 -isystem /usr/include/qt5/QtWidgets -isystem /usr/include/qt5/QtGui -isystem /usr/include/qt5/QtNetwork -isystem /usr/include/qt5/QtCore -Iintermediate -isystem /usr/include/libdrm -Iintermediate -I/usr/lib64/qt5/mkspecs/linux-g++ -o intermediate/qvaboutdialog.o src/qvaboutdialog.cpp
g++ -c -pipe -O2 -fPIC -std=gnu++11 -Wall -W -D_REENTRANT -fPIC -DVERSION=3.0 -DQT_DEPRECATED_WARNINGS -DQT_NO_FOREACH -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I. -isystem /usr/include/qt5 -isystem /usr/include/qt5/QtWidgets -isystem /usr/include/qt5/QtGui -isystem /usr/include/qt5/QtNetwork -isystem /usr/include/qt5/QtCore -Iintermediate -isystem /usr/include/libdrm -Iintermediate -I/usr/lib64/qt5/mkspecs/linux-g++ -o intermediate/qvwelcomedialog.o src/qvwelcomedialog.cpp
g++ -c -pipe -O2 -fPIC -std=gnu++11 -Wall -W -D_REENTRANT -fPIC -DVERSION=3.0 -DQT_DEPRECATED_WARNINGS -DQT_NO_FOREACH -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I. -isystem /usr/include/qt5 -isystem /usr/include/qt5/QtWidgets -isystem /usr/include/qt5/QtGui -isystem /usr/include/qt5/QtNetwork -isystem /usr/include/qt5/QtCore -Iintermediate -isystem /usr/include/libdrm -Iintermediate -I/usr/lib64/qt5/mkspecs/linux-g++ -o intermediate/qvinfodialog.o src/qvinfodialog.cpp
g++ -c -pipe -O2 -fPIC -std=gnu++11 -Wall -W -D_REENTRANT -fPIC -DVERSION=3.0 -DQT_DEPRECATED_WARNINGS -DQT_NO_FOREACH -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I. -isystem /usr/include/qt5 -isystem /usr/include/qt5/QtWidgets -isystem /usr/include/qt5/QtGui -isystem /usr/include/qt5/QtNetwork -isystem /usr/include/qt5/QtCore -Iintermediate -isystem /usr/include/libdrm -Iintermediate -I/usr/lib64/qt5/mkspecs/linux-g++ -o intermediate/qvimagecore.o src/qvimagecore.cpp
src/qvimagecore.cpp: In member function 'void QVImageCore::updateFolderInfo()':
src/qvimagecore.cpp:261:114: error: 'default_random_engine' is not a member of 'std'
         std::shuffle(currentFileDetails.folderFileInfoList.begin(), currentFileDetails.folderFileInfoList.end(), std::default_random_engine(randomSortSeed));
                                                                                                                  ^
Makefile:978: recipe for target 'intermediate/qvimagecore.o' failed
make: *** [intermediate/qvimagecore.o] Error 1

**Comment by jurplel** [comment]
what is your gcc version? It should be a little more recent because it needs c++11 compatability

**Comment by 0pLuS0** [comment]
Slack 14.2 has gcc 5.5.0, so I installed gcc-9.2.0 into /usr/local so I could have both versions in the system.

So now I changed my build script with the export lines for gcc9 like this;

export CC=/usr/local/bin/gcc
export CXX=/usr/local/bin/g++

cd $TMP/$PRGNAM-$VERSION
qmake-qt5
make
make INSTALL_ROOT="$PKG" install

BUT I am still getting the same errors...  :face_with_head_bandage:

**Comment by jurplel** [comment]
This feature was added in gcc 4.5, so I guess the problem isn't there. I tried including <random> which I think will fix it.

**Comment by 0pLuS0** [comment]
Is  gcc 5.5.0 good enough or should I compile against gcc-9.2.0?

This time it compiled using gcc-9.2.0... :)

**Comment by jurplel** [comment]
gcc 5.5.0 should be good enough

**Comment by 0pLuS0** [comment]
Ok gcc-5.5.0 compiled ok.

So when I first ran it, the Welcome screen doesn't close if I click the qview UI, is the first window I noticed with this issue...

![qview](https://user-images.githubusercontent.com/36190223/92675144-41f6b380-f30e-11ea-9ea1-e32138153aa3.jpg)

**Comment by jurplel** [comment]
I see.. Alright thanks, I'll take another look at it. I had an issue for this already so I'll be able to reuse it: #251 
I encountered this before but I wasn't sure if it was my fault or a strange choice by the window manager. Have you ever encountered programs with similar problems before, especially qt ones?

**Comment by 0pLuS0** [comment]
Another issue is if I click on an image that is 1920x1080, wallpapers I use, qviews opens up in a small window size;

1. I start qview from a terminal and it opens.
2. I right click Open and pick and image.
3. The Ui still stays small... hmm (I thought it would enlarge?

![qview_ui_size](https://user-images.githubusercontent.com/36190223/92675385-c5b0a000-f30e-11ea-9bfa-8804bd1bb9de.jpg)

**Comment by jurplel** [comment]
There is an option in preferences to enable the window size changing when opening images

**Comment by 0pLuS0** [comment]
In regards to the original UI issue, no I have never had this problem before.

The settings say;

```Window matches image size: When launching``` I assumed this would mean I could start qview then open an image and it resizes? Hmm that option sounds like that is what it's suppose to do?

I don't have qview assigned as the default app at the moment, but if I right click an image, and choose qview to open it with, then the UI and image is larger...

**Comment by jurplel** [comment]
This is intended behavior. "When launching" means it only matches the image size when the *program* launches. To do it whenever you change an image, change "When launching" to "When opening images".

**Comment by 0pLuS0** [comment]
Sorry, I should of come out and just said, there's an issue here.

If I open qview first, then right click and open an image it won't resize as the preferences as listed.

If I right click on an image and open it with qview then it opens up the image in a larger size.

So just because I open qview first, then right click to bring up the menu and open an image, the UI should still resize and it doesn't that's the problem. This is what I meant by that screen shot I showed of a small image, when it was 1920x1080.

Sorry for any confusion. :)

**Comment by jurplel** [comment]
I think I understand correctly, but that is the intended behavior. By "When launching", I mean only when qview is launched with an image. When you use the command line or open it from a file browser, it will trigger this behavior. When you open the program and navigate to the open dialog, that will not resize the window.

**Comment by 0pLuS0** [comment]
I see now with it on ```When launching``` this is what happens.

With it on ```When opening images``` then it will change when I first open qview and then an image.

I am either getting thick in the head, or the wording/behaviour seems odd to me... hmm

Reading the wording;

```Window matches image size: When launching``` just reads to me as;

However I launch qview, starting it first, or clicking on an image, this option above sounds like it works either way and it doesn't, that is the confusion for me here... Make sense?

THANKS

**Comment by jurplel** [comment]
I can see how the wording might be confusing. This functionality was originally made because other programs had similar behavior.

**Comment by 0pLuS0** [comment]
I just left it so the window stays a larger size... 

THANKS

P.S. Great app, keep up the great work, greatly appreciated! Man that's a lot of greats! LOL... :)

**Comment by jurplel** [comment]
Hah, thank you! Nice talking to you and thanks for bringing my attention to the issue with the close button.

**Comment by 0pLuS0** [comment]
@jurplel 

Please let me know when the windows should close now, so I can test?

Thanks

**Comment by 0pLuS0** [comment]
Ok thanks...

**Comment by jurplel** [comment]
Well it's been a while but I finally got around to looking at this again and I just realized that I misinterpreted the issue. I thought you were talking about a missing close button, but instead it seems that you are saying that the welcome window should close when qView is clicked on. The way it works now is the intended behavior, where the welcome window has to be closed first. Am I understanding correctly?
